src/service/extensions-impl.js

```diff
@@ -242,32 +242,56 @@ export class Extensions {
    * @return {!Promise<!ExtensionDef>}
    */
   reloadExtension(extensionId) {
-    const attribute = isTemplateExtension(extensionId)
-      ? 'custom-template'
-      : 'custom-element';
-    // The :not is an extra prevention of recursion because it will be
-    // added to script tags that go into the code path below.
-    const oldScriptElement = this.win.document.head./*OK*/ querySelector(
-      `[${attribute}="${extensionId}"]:not([i-amphtml-inserted])`
-    );
-    devAssert(
-      oldScriptElement,
-      'Expected to find script for extension: %s',
-      extensionId
+    // Ignore inserted script elements to prevent recursion.
+    const extension = this.getExtensionScript_(
+      extensionId,
+      /* includeInserted */ false
     );
-
+    devAssert(extension, 'Cannot find script for extension: %s', extensionId);
+    const {el, attr} = extension;
     // "Disconnect" the old script element and extension record.
     const holder = this.extensions_[extensionId];
     if (holder) {
       devAssert(!holder.loaded && !holder.error);
       delete this.extensions_[extensionId];
     }
-    oldScriptElement.removeAttribute(attribute);
-    oldScriptElement.setAttribute('i-amphtml-loaded-new-version', extensionId);
-    const urlParts = parseExtensionUrl(oldScriptElement.src);
+    if (attr) {
+      el.removeAttribute(attr);
+    }
+    el.setAttribute('i-amphtml-loaded-new-version', extensionId);
+    // TODO(wg-runtime): preloadExtension() calls getExtensionScript_() again
+    // to check for existence of the script, which is redundant.
+    const urlParts = parseExtensionUrl(el.src);
     return this.preloadExtension(extensionId, urlParts.extensionVersion);
   }
 
+  /**
+   * Returns the extension <script> element and attribute for the given
+   * extension ID, if it exists. Otherwise, returns null.
+   * @param {string} extensionId
+   * @param {boolean=} includeInserted If true, includes script elements that
+   *   are inserted by the runtime dynamically. Default is true.
+   * @return {?{el: !Element, attr: (string|undefined)}}
+   * @private
+   */
+  getExtensionScript_(extensionId, includeInserted = true) {
+    const attr = this.attributeForExtension_(extensionId);
+    const modifier = includeInserted ? '' : ':not([i-amphtml-inserted])';
+    const {head} = this.win.document;
+    let el = head./*OK*/ querySelector(
+      `script[${attr}="${extensionId}"]` + modifier
+    );
+    if (el) {
+      return {el, attr};
+    }
+    // Some extensions don't have an attribute e.g. amp-viewer-integration.
+    el = head./*OK*/ querySelector(`script[src*="${extensionId}-"]` + modifier);
```

Comment:
```suggestion
    el = head./*OK*/ querySelector(`script[src*="/${extensionId}-"]` + modifier);
```

The slash was important. 😛